utils_cv/tracking/plot.py

```diff
@@ -1,12 +1,132 @@
+from .model import _get_frame
+
+
+def plot_single_frame(
+    results: Dict[int, List[TrackingBbox]], input_video: str, frame_id: int
+) -> None:
+    """ 
+    Plot the bounding box and id on a wanted frame. Display as image to front end. 
+
+    Args:
+        results: dictionary mapping frame id to a list of predicted TrackingBboxes
+        input_video: path to the input video
+        frame_id: frame_id for frame to show tracking result
+    """
+    results = OrderedDict(sorted(results.items()))
+
+    # Assign bbox color per id
+    unique_ids = list(
+        set([bb.track_id for frame in results.values() for bb in frame])
+    )
+    color_map = assign_colors(unique_ids)
+
+    # Get frame from video
+    im = _get_frame(input_video, frame_id)
+
+    # Extract tracking results for wanted frame, and draw bboxes+tracking id, display frame
+    cur_tracks = results[frame_id]
+    if len(cur_tracks) > 0:
+        im = draw_boxes(im, cur_tracks, color_map)
+    im = Image.fromarray(im)
+    IPython.display.display(im)
+
+
+def play_video(
+    results: Dict[int, List[TrackingBbox]], input_video: str
+) -> None:
+    """ 
+     Plot the predicted tracks on the input video. Displays to front-end as sequence of images stringed together in a video.
+
+    Args:
+        results: dictionary mapping frame id to a list of predicted TrackingBboxes
+        input_video: path to the input video        
+    """
+
+    results = OrderedDict(sorted(results.items()))
+
+    # assign bbox color per id
+    unique_ids = list(
+        set([bb.track_id for frame in results.values() for bb in frame])
+    )
+    color_map = assign_colors(unique_ids)
+
+    # read video and initialize new tracking video
+    video_reader = decord.VideoReader(input_video)
+
+    # set up ipython jupyter display
+    d_video = IPython.display.display("", display_id=1)
+
+    # Read each frame, add bbox+track id, display frame
+    for frame_idx in range(len(results)):
+        im = video_reader.next().asnumpy()
+        cur_tracks = results[frame_idx]
+        if len(cur_tracks) > 0:
+            cur_image = draw_boxes(im, cur_tracks, color_map)
+
+        f = io.BytesIO()
+        im = Image.fromarray(im)
+        im.save(f, "jpeg")
+        d_video.update(IPython.display.Image(data=f.getvalue()))
+        sleep(0.000001)
+
+
+def write_video(
+    results: Dict[int, List[TrackingBbox]], input_video: str, output_video: str
+) -> None:
+    """ 
+    Plot the predicted tracks on the input video. Write the output to {output_path}.
+
+    Args:
+        results: dictionary mapping frame id to a list of predicted TrackingBboxes
+        input_video: path to the input video
+        output_video: path to write out the output video
+    """
+    results = OrderedDict(sorted(results.items()))
+    # read video and initialize new tracking video
+    video = cv2.VideoCapture()
+    video.open(input_video)
+
+    image_width = int(video.get(cv2.CAP_PROP_FRAME_WIDTH))
```

Comment:
As pointed out in my last code review, I believe these 4 lines are not needed anymore and can get deleted

Reply:
These 4 lines are still needed for this `write_video()` function? `image_width`, `image_height` are called by `cv2.VideoWriter` in line 109. I did delete similar lines which were redundant in `plot_single_frame()` function.

Reply:
My bad, i see it. Maybe rename to im_width, and im_height, but looks good then